packages/medusa/src/services/customer.ts

```diff
@@ -202,23 +167,25 @@ class CustomerService extends BaseService {
    * @param {Object} config - the config object containing query settings
    * @return {Promise<Customer>} the customer document.
    */
-  async retrieve(customerId, config = {}) {
-    const customerRepo = this.manager_.getCustomRepository(
-      this.customerRepository_
-    )
+  async retrieve(
```

Comment:
suggestion: For the next three methods, I would suggest something around the following lines:
```ts
private async retrieve_(
  selector: Selector<Customer>,
  config: FindConfig<Customer> = {}
): Promise<Customer | never> {
  const transactionManager = this.transactionManager_ ?? this.manager_

  const customerRepo = transactionManager.getCustomRepository(this.customerRepository_)

  const query = buildQuery(selector, config)
  const customer = await customerRepo.findOne(query)

  if (!customer) {
    const selectorConstraints = Object
      .entries(selector)
      .map(([key, value] => `${key}: ${value}`)).join(', ')
    throw new MedusaError(
      MedusaError.Types.NOT_FOUND,
      `Customer with ${selectorConstraints} was not found`
    )
  }

  return customer
}

async retrieveByEmail(
  email: string,
  config: FindConfig<Customer> = {}
): Promise<Customer | never> {
  return await this.atomicPhase_(async () => {
    return await this.retrieve_({ email: email.toLowerCase() }, config)
  })
}

async retrieveByPhone(
  phone: string,
  config: FindConfig<Customer> = {}
): Promise<Customer | never> {
  return await this.atomicPhase_(async () => {
    return await this.retrieve_({ phone }, config)
  })
}
```